azure/compute/vm/azurevm.go

```diff
@@ -34,20 +34,20 @@ func newVM(ctx *pulumi.Context, options ...func(*Params) error) (commonvm.VM, er
 		return nil, err
 	}
 
-	_, publicIP, _, err := compute.NewLinuxInstance(
+	vm, publicIP, _, err := compute.NewLinuxInstance(
 		env,
 		env.CommonNamer.ResourceName(params.common.InstanceName),
 		params.common.ImageName,
 		params.common.InstanceType,
 		pulumi.StringPtr(params.common.UserData),
 	)
-
 	if err != nil {
 		return nil, err
 	}
 
 	return commonvm.NewGenericVM(
 		params.common.InstanceName,
+		vm,
 		&env,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Do we still need to pass the environment since we pass the vm ?